apiVersion: openapi/v1alpha1
kind: Radar
metadata:
    name: "31505061"
    labels:
        datastore_id: "4988970542825472"
data:
    classification: Serious Bug
    created: "2017-04-07T18:28:07.43291Z"
    description: "Area:\r\nFoundation\r\n\r\nSummary:\r\nIf the 1st background URLSessionDownloadTask resuming complete download successfully everything is fine.\r\n\r\nHowever, if the 1st trial is interrupted, say because the app is skilled again, the following attempts to resume the download can still succeed, but the `location` argument passed in urlSession(_:downloadTask:didFinishDownloadingTo:) is wrong which causes the downloaded file cannot be correctly used. Notice that the downloaded file is there in the cache dir, just we cannot get its location in the callback.\r\n\r\nSteps to Reproduce:\r\n1. On your testing device ,turn on system Settings > Developer > Network Link Conditioner with Edge profile, in case your network is too good to leave you any time to do the operations.\r\n\r\n2. Launch sample app from Xcode.\r\n3. Kill the app by Stop in Xcode after you see some % download progress in console logs.\r\n4. Relaunch sample app from Xcode.\r\n5. Wait until the download to complete.\r\n6. Read the console log to see the good `location`.\r\n\r\n7. Repeat 2~4.\r\n8. Kill the app again by Stop in Xcode after you see more % download progress in console logs.\r\n9. Repeat 4~5.\r\n10. Read the console log to see the wrong `location` and correct url.\r\n\r\nCode: https://cloudup.com/cC6YM6SXRDf"
    email: an00na@gmail.com
    modified: "2017-04-07T18:28:07.43321Z"
    number: "31505061"
    number_intvalue: 31505061
    originated: 07-Apr-2017 01:26 PM
    parent_number: '&{NULL_VALUE}'
    product: iOS + SDK
    product_version: "10.3"
    reproducible: Always
    resolved: ""
    status: Open
    title: Background URLSessionDownloadTask bug
